crates/forge/src/fuzzer.rs

```diff
@@ -1,16 +1,124 @@
 use rand::{Rng, SeedableRng};
+use std::ops::{Shl, Shr, Sub};
 
+#[derive(Debug, Clone)]
+enum CairoType {
+    U8,
+    U16,
+    U32,
+    U64,
+    U128,
+    U256,
+    Felt252,
+}
+
+trait Argument {
+    fn low(&self) -> BigUint;
+    fn high(&self) -> BigUint;
+    fn gen(&self, rng: &mut StdRng) -> Vec<Felt252>;
+    fn min(&self) -> Vec<Felt252>;
+    fn max(&self) -> Vec<Felt252>;
+}
+
+impl Argument for CairoType {
+    fn low(&self) -> BigUint {
+        BigUint::zero()
+    }
+
+    fn high(&self) -> BigUint {
+        match self {
+            CairoType::U8 => BigUint::from(u8::MAX),
+            CairoType::U16 => BigUint::from(u16::MAX),
+            CairoType::U32 => BigUint::from(u32::MAX),
+            CairoType::U64 => BigUint::from(u64::MAX),
+            CairoType::U128 => BigUint::from(u128::MAX),
+            CairoType::U256 => {
+                let max = BigUint::from(1_u32);
+                let max = max.shl(256);
+                max - BigUint::one()
+            }
+            CairoType::Felt252 => Felt252::prime().sub(BigUint::one()),
+        }
+    }
+
+    fn gen(&self, rng: &mut StdRng) -> Vec<Felt252> {
+        match self {
+            CairoType::U8
+            | CairoType::U16
+            | CairoType::U32
+            | CairoType::U64
+            | CairoType::U128
+            | CairoType::Felt252 => {
+                vec![Felt252::from(
+                    rng.gen_biguint_range(&self.low(), &self.high()),
+                )]
+            }
+            CairoType::U256 => {
+                let val = rng.gen_biguint_range(&self.low(), &self.high());
+                u256_to_felt252(val)
+            }
+        }
+    }
+
+    fn min(&self) -> Vec<Felt252> {
+        match self {
+            CairoType::U8
+            | CairoType::U16
+            | CairoType::U32
+            | CairoType::U64
+            | CairoType::U128
+            | CairoType::Felt252 => vec![Felt252::from(self.low())],
+            CairoType::U256 => vec![Felt252::from(self.low()), Felt252::from(self.low())],
+        }
+    }
+
+    fn max(&self) -> Vec<Felt252> {
+        match self {
+            CairoType::U8
+            | CairoType::U16
+            | CairoType::U32
+            | CairoType::U64
+            | CairoType::U128
+            | CairoType::Felt252 => vec![Felt252::from(self.high())],
+            CairoType::U256 => u256_to_felt252(self.high()),
+        }
+    }
+}
+
+fn u256_to_felt252(val: BigUint) -> Vec<Felt252> {
+    let low = val.mod_floor(&BigUint::from(2_u32).pow(128));
+    let high = val.shr(128);
+    vec![Felt252::from(low), Felt252::from(high)]
+}
+
+impl CairoType {
+    fn from_name(name: &str) -> Result<Self> {
+        match name {
+            "u8" => Ok(Self::U8),
+            "u16" => Ok(Self::U16),
+            "u32" => Ok(Self::U32),
+            "u64" => Ok(Self::U64),
+            "u128" => Ok(Self::U128),
+            "u256" | "core::integer::u256" => Ok(Self::U256),
```

Comment:
I assume it cannot happen in other cases as those are builtin types, right?

Reply:
I think so. Actually `u256` is I think always represented as `"core::integer::u256"` only. But I've added `u256` for consistency sake + it doesn't hurt us to support that as well.